azurerm/internal/services/kusto/kusto_cluster_resource.go

```diff
@@ -95,6 +95,34 @@ func resourceArmKustoCluster() *schema.Resource {
 				},
 			},
 
+			"optimized_auto_scale": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"version": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+						"enabled": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
```

Comment:
Since it appears (below) that these fields are configurable - should we make these configurable (or remove them)? Since this isn't ForceNew - at the moment it appears there's no way to move from enabled -> disabled

Reply:
`enabled` will be changed to be configurable. I also have to test whether ForceNew is required if this property is configurable. meaning of version needs also to be evaluated.

Reply:
if version is unclear let's remove this for now and default it to version 1 - we can always expose it later if it proves to be useful to configure :)